uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/models/instances/leave-request-instance.js

```diff
@@ -247,9 +269,53 @@ define([
          * @param {string} key - The property name
          */
         toAPIFilter: function (result, __, key) {
-          if (!_.includes(['balance_change', 'dates', 'comments'], key)) {
+          if (!_.includes(['balance_change', 'dates', 'comments', 'uploader', 'files'], key)) {
             result[key] = this[key];
           }
+        },
+
+        /**
+         * Loads file attachments associated with this leave request
+         *
+         * @return {Promise}
+         */
+        loadAttachments: function () {
+          return LeaveRequestAPI.getAttachments(this.id)
+            .then(function (attachments) {
+              this.files = attachments;
+            }.bind(this));
+        },
+
+        /**
+         * Sets the flag to mark file for deletion. The file is not yet deleted
+         * from the server.
+         */
+        deleteAttachment: function (attachmentID) {
+          this.files = this.files.map(function (file) {
```

Comment:
Point no 1 is more optimal as we are not re assigning the variable.
Also `map` should only be used when we need a new copy of the array(this is more cpu extensive), but here we just need to change value of a flag so `each`should be enough and more optimal.